src/DirIter.cc

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <filesystem>
+
+#include "ignition/common/Filesystem.hh"
+
+namespace fs = std::filesystem;
+
+namespace ignition
+{
+namespace common
+{
+
+class DirIterPrivate
+{
+  public: fs::directory_iterator it;
+};
+
+DirIter::DirIter():
+  dataPtr(std::make_unique<DirIterPrivate>())
+{
+  this->dataPtr->it = fs::directory_iterator();
+}
+
+DirIter::DirIter(const std::string &_in):
+  DirIter()
+{
+  try {
+    this->dataPtr->it = fs::directory_iterator(_in);
+  }
+  catch (const fs::filesystem_error &ex)
+  {
+  }
```

Comment:
Error msgs ?

Reply:
This doesn't really need an error message.  If the path is invalid, then this will be set to the `::end()` iterator, so any loops will end.